pkg/controllers/monitoring/monitoring-stack/conditions.go

```diff
@@ -0,0 +1,121 @@
+package monitoringstack
+
+import (
+	monv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	stack "github.com/rhobs/observability-operator/pkg/apis/monitoring/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	monitoringStackAvailable       = "MonitoringStackAvailable"
+	monitoringStackReconciled      = "MonitoringStackReconciled"
+	prometheusNotAvailable         = "PrometheusNotAvailable"
+	prometheusNotReconciled        = "PrometheusNotReconciled"
+	cannotReadPrometheusConditions = "Cannot read Prometheus status conditions"
+)
+
+type conditions map[stack.ConditionType]*stack.Condition
+
+type conditionsMap struct {
```

Comment:
not sure if we need this additional type? couldn't we use `conditions` directly?

Reply:
I think we need the extra type, because we can't do receiver like `func (cm *conditions) UpdateReconciled`only `func (cm conditions) UpdateReconciled`, which means copying the map right...pls correct me if I am wrong

Reply:
`func (cm conditions) UpdateReconciled(...)` would update the map despite being a value receiver I believe (see https://stackoverflow.com/questions/49176090/map-as-a-method-receiver)